src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -2955,7 +2956,7 @@ void c_typecheck_baset::typecheck_expr_binary_arithmetic(exprt &expr)
     is_number(o_type1))
   {
     // convert op1 to the vector type
-    op1.make_typecast(o_type0);
+    op1 = typecast_exprt::conditional_cast(op1, o_type0);
     expr.type() = o_type0;
```

Comment:
A conditional_cast appears to be wrong here. When we are in this branch, the cast is always needed. Simply use the constructor.

Reply:
Done.